hack/postrelease/golang/calico_oss_test.go

```diff
@@ -0,0 +1,132 @@
+package calico_oss_test
+
+import (
+	"calico_postrelease/pkg/container"
+	"calico_postrelease/pkg/registry"
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var release_tag = "v3.27.4"
+
+var expected_arches = []string{"amd64", "arm64", "s390x", "ppc64le"}
+
+var expected_images = []string{
+	"calico/node",
+	"calico/ctl",
+	"calico/apiserver",
+	"calico/typha",
+	"calico/cni",
+	"calico/kube-controllers",
+	// "calico/upgrade",
+	"calico/flannel-migration-controller",
+	"calico/dikastes",
+	// "calico/pilot-webhook",
+	"calico/pod2daemon-flexvol",
+	"calico/csi",
+}
+
+// Commenting these out because we don't currently tag the
+// individual windows images. Should we?
+// var expected_windows_tags = []string{
+// 	"windows-ltsc2022",
+// 	"windows-1809",
+// }
+
+var expected_windows_images = []string{
+	"calico/cni-windows",
+	"calico/node-windows",
+}
+
+func TestGolang(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Calico OSS Postrelease Test Suite")
+}
+
+var _ = Describe(
+	"Validate Container Image Publishing",
+	func() {
+		var operatorImage container.Image
+		var quayRegistry registry.QuayRegistry = registry.NewQuayRegistry()
+		var gcrRegistries = []registry.GCRRegistry{
+			registry.NewGCRRegistry("gcr.io"),
+			registry.NewGCRRegistry("us.gcr.io"),
+			registry.NewGCRRegistry("asia.gcr.io"),
+			registry.NewGCRRegistry("us.gcr.io"),
+		}
+		var imagesToTest = make([]container.Image, 0)
+
+		for _, image_name := range expected_images {
+			for _, arch_name := range expected_arches {
+				image_name := image_name
+				arch_name := arch_name
+				var containerImage = container.Image{
+					Name: image_name,
+					Tag:  fmt.Sprintf("%s-%s", release_tag, arch_name),
```

Comment:
While this checks for arch-tagged images, does this script check that the main image is a manifest image which has pointers to the correct specific arch images?

Reply:
fwiw, regctl can do this

Reply:
It doesn't check that, no. It will, but it doesn't.